readthedocs/projects/forms.py

```diff
@@ -1025,8 +1026,35 @@ def clean_domain(self):
             if invalid_domain and domain_string.endswith(invalid_domain):
                 raise forms.ValidationError(f"{invalid_domain} is not a valid domain.")
 
+        if self._has_cname(domain_string):
+            raise forms.ValidationError(
+                _(
+                    "This domain has a CNAME record. "
+                    "Please remove it before adding the domain.",
+                ),
+            )
+
         return domain_string
 
+    def _has_cname(self, domain):
+        """
+        Check if a domain has a CNAME record.
+
+        We make use of `dig` to check if a domain has a CNAME record,
+        if it doesn't return an empty string, it means it has a CNAME record.
+        """
+        try:
+            result = subprocess.run(
+                ["dig", "-t", "CNAME", "+short", domain],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                timeout=1,
+                check=True,
+            )
+        except (subprocess.TimeoutExpired, subprocess.CalledProcessError):
+            return False
+        return bool(result.stdout.strip())
```

Comment:
Pretty sure we don't want to fail this on _any_ CNAME record. We only want to fail it on chained CNAME records, right? It feels weird to make users delete an existing valid CNAME record that already points to RTD? 

I guess that stops the issue with domain hacking of an existing domain more explicitly, but also makes it a pain to move a domain from 1 project to another, delete and recreate a domain, and other types of pretty common workflows. 

I feel this approach is too aggressive, and we just want to blocked chained domains (eg. DNS queries that return `>1` line in this query)?

Reply:
I think these are really good arguments _against_ the approach proposed in this PR, and I agree it's too aggressive. In particular, when the ops team is decoupled with the technical writers team; if the technical writers team has to contact the ops team for this multiple times, it would be a mess.